cmd/notation/internal/display/metadata/json/inspect.go

```diff
@@ -0,0 +1,204 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/signature"
+	"github.com/notaryproject/notation-go/plugin/proto"
+	"github.com/notaryproject/notation/cmd/notation/internal/output"
+	"github.com/notaryproject/notation/internal/envelope"
+	"github.com/notaryproject/tspclient-go"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type inspectOutput struct {
+	MediaType  string `json:"mediaType"`
+	Signatures []Signature
+}
+
+// Signature is the signature envelope for printing in human readable format.
+type Signature struct {
+	MediaType             string             `json:"mediaType"`
+	Digest                string             `json:"digest,omitempty"`
+	SignatureAlgorithm    string             `json:"signatureAlgorithm"`
+	SignedAttributes      map[string]any     `json:"signedAttributes"`
+	UserDefinedAttributes map[string]string  `json:"userDefinedAttributes"`
+	UnsignedAttributes    map[string]any     `json:"unsignedAttributes"`
+	Certificates          []Certificate      `json:"certificates"`
+	SignedArtifact        ocispec.Descriptor `json:"signedArtifact"`
+}
+
+// Certificate is the certificate information for printing in human readable
+// format.
+type Certificate struct {
+	SHA256Fingerprint string    `json:"SHA256Fingerprint"`
+	IssuedTo          string    `json:"issuedTo"`
+	IssuedBy          string    `json:"issuedBy"`
+	Expiry            time.Time `json:"expiry"`
+}
+
+// Timestamp is the timestamp information for printing in human readable.
+type Timestamp struct {
+	Timestamp    string        `json:"timestamp,omitempty"`
+	Certificates []Certificate `json:"certificates,omitempty"`
+	Error        string        `json:"error,omitempty"`
+}
+
+// InspectHandler is the handler for inspecting metadata information and
+// rendering it in JSON format.
+type InspectHandler struct {
+	printer *output.Printer
+
+	output inspectOutput
+}
+
+// NewInspectHandler creates a new InspectHandler.
+func NewInspectHandler(printer *output.Printer) *InspectHandler {
+	return &InspectHandler{
+		printer: printer,
+	}
+}
+
+// OnReferenceResolved sets the artifact reference and media type for the
+// handler.
+//
+// the reference is no-op for this handler.
+func (h *InspectHandler) OnReferenceResolved(reference, mediaType string) {
+	if h.output.MediaType == "" {
+		h.output.MediaType = mediaType
+	}
+}
+
+// InspectSignature inspects a signature to get it ready to be rendered.
+func (h *InspectHandler) InspectSignature(digest string, envelopeMediaType string, sigEnvelope signature.Envelope) error {
+	sig, err := newSignature(digest, envelopeMediaType, sigEnvelope)
+	if err != nil {
+		return err
+	}
+	h.output.Signatures = append(h.output.Signatures, *sig)
```

Comment:
What if the artifact has no signatures?

Basically, I am find the related code for
```go
output := inspectOutput{MediaType: manifestDesc.MediaType, Signatures: []signatureOutput{}}
```

Reply:
Updated the constructor to initialize the signature slice.